### PR TITLE
docs: Update APIDataset docs and refactor

### PR DESCRIPTION
**Pull request by noklam**
## Description

https://github.com/kedro-org/kedro-plugins/pull/180 was auto-merged, this is a continuation of it.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Woops, sorry @noklam - saw your approval but not your comment, apologies 🙏🏽

**Review by astrojuanlu — APPROVED** [review]

**Review by ankatiyar — APPROVED** [review]

File: kedro-datasets/RELEASE.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 Many thanks to the following Kedroids for contributing PRs to this release:
 
 * [BrianCechmanek](https://github.com/BrianCechmanek)
+* [McDonnellJoseph](https://github.com/McDonnellJoseph)
 
 # Release 1.2.1:
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply kedro-datasets/RELEASE.md:20]: So, @ankatiyar you mean deleting the line `Release 1.2.1` and merging the contents?

[the remark this answers is not in the document]
Comment (ankatiyar) [reply kedro-datasets/RELEASE.md:20]: Yes! I don't think we've released 1.2.1 yet.

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply kedro-datasets/RELEASE.md]: Done, please double check 🙏🏽 @ankatiyar

[the remark this answers is not in the document]
Comment (ankatiyar) [reply kedro-datasets/RELEASE.md]: Looks good @astrojuanlu, thanks! 😄

**Pull request merged**